src/rdo.rs

```diff
@@ -782,6 +793,29 @@ fn luma_chroma_mode_rdo<T: Pixel>(
   let is_chroma_block =
     has_chroma(tile_bo, bsize, xdec, ydec, fi.sequence.chroma_sampling);
 
+  if !luma_mode_is_intra {
+    let ref_mvs = if mv_stack.len() > 0 {
+      [mv_stack[0].this_mv, mv_stack[0].comp_mv]
+    } else {
+      [MotionVector::default(); 2]
+    };
+
+    if (luma_mode == PredictionMode::NEWMV
+      || luma_mode == PredictionMode::NEW_NEWMV
+      || luma_mode == PredictionMode::NEW_NEARESTMV)
+      && !mv_in_range(mvs[0], ref_mvs[0])
+    {
+      return;
```

Comment:
Would there be case that all inter mode candidates, i.e. MVs tested by "sorted.iter().take(num_modes_rdo).for_each( ...
...
luma_chroma_mode_rdo( )..."
are out of range and return, thus all inter modes are dropped?

To avoid such case, are we able to revise motion_estimation() clip the MV to valid range, if the clipped is not completely bad choice.

Reply:
As far as I can tell, constraining the motion-estimation is necessary since libaom enforces bounds on the absolute motion vector values. I have added this constraint.